be more specific about how unversioned vendored deps work (#1061)
fossas/fossa-cli: docs/features/vendored-dependencies.md

If it is a path to an archive, then we recursively unarchive and scan the archive. If it is a directory, then we scan the directory and recursively unarchive and scan any archives contained in the directory.

> If the version is not specified, then we calculate the version by calculating a hash of the contents of the archive or directory. This is often what you actually want, as it means that the version will automatically change when the contents of the vendored dependency change. It also avoids conflicts across your organization when two different projects contain a vendored dependency with the same name and version, as described in [Vendored Dependency Names and Scope](#vendored-dependency-names-and-scope).

Reviewer: [nit] It's generally recommended to avoid speaking to the user or in first person in technical writing.
 
Consider this instead:
```suggestion
If the version is not specified, then FOSSA CLI calculates the version by generating a hash of the contents of the archive or directory. This is often desired, as it means that the version automatically changes when the contents of the vendored dependency change. It also avoids conflicts across an organization when two different projects contain a vendored dependency with the same name and version, as described in [Vendored Dependency Names and Scope](#vendored-dependency-names-and-scope).
```

> If the version is not specified, then we calculate the version by calculating a hash of the contents of the archive or directory. This is often what you actually want, as it means that the version will automatically change when the contents of the vendored dependency change. It also avoids conflicts across your organization when two different projects contain a vendored dependency with the same name and version, as described in [Vendored Dependency Names and Scope](#vendored-dependency-names-and-scope).

Reviewer 2: ```suggestion
If the version is not specified, we calculate the version by calculating a hash of the contents of the archive or directory. This is often what you actually want, as it means that the version will automatically change when the contents of the vendored dependency change. It also avoids conflicts across your organization when two different projects contain a vendored dependency with the same name and version, as described in [Vendored Dependency Names and Scope](#vendored-dependency-names-and-scope).
```

  ]
}
```

## Vendored Dependency Names and Scope

The name of a vendored dependency is scoped to an organization.

> This means that if two different projects in your organization have the same name and version, they will be treated as the same dependency by FOSSA. The one that was scanned first will be used by FOSSA. Here's an example.

Reviewer 2: You should make this example a subsection of the "Vendored Dependency..." section so that it is easier to link if other users have the same issue.

I think something like the following would be a simple change.
```
### Conflicts with dependency names
Problem
Example
```

Author: I think that's a good idea in general, but that makes the parent section pretty much empty in this case. I'm just going to leave it as is, as I don't think there's much benefit to pulling it out

> Project A defines a vendored dependency like this in its fossa-deps.yml, and the contents of that vendored dependency has an MIT license:

Reviewer 2: ```suggestion
Project A defines a vendored dependency like this in its fossa-deps.yml, and the contents of that vendored dependency have an MIT license:
```

```yaml
vendored-dependencies:
- name: Django
  path: vendor/Django-3.4.16.zip # path can be either a file or a folder.
  version: "3.4.16" # revision will be set to the MD5 hash of the filepath if left unspecified.
```

Project B has exactly the same contents in `fossa-deps.yml`, but the contents of Project B's vendored dependency have an Apache 2.0 license.

If Project A is scanned first, then both Project A and Project B will report that their vendored dependency has an MIT license.

If Project B is scanned first, then both Project A and Project B will report that their vendored dependency has an Apache 2.0 license.

This can cause unexpected behavior, and we are working on changing this so that vendored dependencies are scoped to projects rather than organizations.

> Our suggested workaround is to not set a version on your vendored dependency. When you omit the version, FOSSA will calculate a version based on the contents of the vendored dependency, thus avoiding any conflicts. This also has the added benefit of automatically changing the version when the contents of the vendored dependency change.

Reviewer: [nit] Same as above. Consider something more like this:
<hr>

This means that if two different projects in an organization have the same name and version, they are treated as the same dependency by FOSSA, and the one that was scanned first is reported. As an example:

Project A defines a vendored dependency like this in its `fossa-deps.yml`, and the contents of that vendored dependency contain an MIT license:

```yaml
vendored-dependencies:
- name: Django
  path: vendor/Django-3.4.16.zip
  version: "3.4.16"
```

Project B has exactly the same contents in `fossa-deps.yml`, but the contents of Project B's vendored dependency contain an Apache 2.0 license.

If Project A is scanned first, then both Project A and Project B report that their vendored dependency has an MIT license.

If Project B is scanned first, then both Project A and Project B report that their vendored dependency has an Apache 2.0 license.

This can cause unexpected behavior. (Note: FOSSA is working on changing this so that vendored dependencies are scoped to projects rather than organizations.)

The suggested workaround is to not set a version in the vendored dependency entry. When the version is omitted, FOSSA calculates a version based on the contents of the vendored dependency, thus avoiding any conflicts. This also has the added benefit of automatically changing the version when the contents of the vendored dependency change.
